iroh/src/config.rs

```diff
@@ -163,6 +262,144 @@ impl Config {
     }
 }
 
+/// Environment for CLI and REPL
+///
+/// This is cheaply cloneable and has interior mutability. If not running in the console
+/// environment, [Self::set_doc] and [Self::set_author] will lead to an error, as changing the
+/// environment is only supported within the console.
+#[derive(Clone, Debug)]
+pub struct ConsoleEnv(Arc<Mutex<ConsoleEnvInner>>);
```

Comment:
it seems that most commands will be reading from this, so maybe worth using an `RwLock`?

Reply:
Fixed